flink-table-store-core/src/test/java/org/apache/flink/table/store/table/SchemaEvolutionTableTestBase.java

```diff
@@ -0,0 +1,301 @@
+            BiConsumer<R, Map<Long, TableSchema>> secondChecker,
+            List<String> primaryKeyNames,
+            Configuration tableConfig,
+            Function<Map<Long, TableSchema>, FileStoreTable> createFileStoreTable)
+            throws Exception {
+        Map<Long, TableSchema> tableSchemas = new HashMap<>();
+        tableSchemas.put(
+                0L,
+                new TableSchema(
+                        0,
+                        SCHEMA_0_FIELDS,
+                        5,
+                        PARTITION_NAMES,
+                        primaryKeyNames,
+                        tableConfig.toMap(),
+                        ""));
+        FileStoreTable table = createFileStoreTable.apply(tableSchemas);
+        TableWrite write = table.newWrite("user");
+        TableCommit commit = table.newCommit("user");
+
+        write.write(
+                GenericRowData.of(
+                        StringData.fromString("S001"),
+                        1,
+                        11,
+                        StringData.fromString("S11"),
+                        111L,
+                        StringData.fromString("S111")));
+        write.write(
+                GenericRowData.of(
+                        StringData.fromString("S002"),
+                        2,
+                        12,
+                        StringData.fromString("S12"),
+                        112L,
+                        StringData.fromString("S112")));
+        write.write(
+                GenericRowData.of(
+                        StringData.fromString("S003"),
+                        1,
+                        13,
+                        StringData.fromString("S13"),
+                        113L,
+                        StringData.fromString("S113")));
+        commit.commit(0, write.prepareCommit(true, 0));
+
+        write.write(
+                GenericRowData.of(
+                        StringData.fromString("S004"),
+                        1,
+                        14,
+                        StringData.fromString("S14"),
+                        114L,
+                        StringData.fromString("S114")));
+        write.write(
+                GenericRowData.of(
+                        StringData.fromString("S005"),
+                        2,
+                        15,
+                        StringData.fromString("S15"),
+                        115L,
+                        StringData.fromString("S115")));
+        write.write(
+                GenericRowData.of(
+                        StringData.fromString("S006"),
+                        2,
+                        16,
+                        StringData.fromString("S16"),
+                        116L,
+                        StringData.fromString("S116")));
+        commit.commit(0, write.prepareCommit(true, 0));
+        write.close();
+        R result = firstChecker.apply(tableSchemas);
+
+        tableSchemas.put(
```

Comment:
Add comments to explain fields removing and fields adding, this can help to understand the test code.